### PR TITLE
core(proto): clarify deprecated state of EmulatedFormFactor enum

### PR DESCRIPTION
**Pull request by paulirish**
small tweak to the proto changes from #11779

the enum isn't actually deprecated..

I would have made two separate Enums, as semantically they're _somewhat_ different.. but, you can't have two enum values that aren't unique.  example:

image

So sharing the same enum made sense.

Anyway, lots of comments.

### CODE REVIEW

**Review by patrickhulce — APPROVED** [review]

File: proto/lighthouse-result.proto [hunk 1]
```diff
@@ -127,15 +127,18 @@ message LighthouseResult {
   // Message containing the configuration settings for the LH run
   message ConfigSettings {
 
-    // The possible form factors an audit can be run in
-    enum EmulatedFormFactor { // Deprecated
+    // The possible form factors an audit can be run in.
+    // This enum served the emulated_form_factor field, but in v7, that field was deprecated.
+    // Meanwhile in v7, the form_factor field was added and has a mobile/desktop enum, so this
+    // "legacy-ish" enum gets reused.
+    // https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md#changes-made-in-v7
```
Comment (brendankenny) [inline proto/lighthouse-result.proto:134]: does any of this matter to consumers of the proto? It's an enum that's semantically used for the same thing as before except `none` was deprecated as an option, and that's the part that's _not_ explained by the comment :P

Maybe (some version of) this comment makes more sense down on `emulated_form_factor` or the new `form_factor`?

**Review by connorjclark — COMMENTED** [review]
Can you also add the `form_factor` field we need? since you're already here ;)

@exterkamp and I just determined that as necessary.

**Review by connorjclark — APPROVED** [review]

File: proto/lighthouse-result.proto (hunk 1 above)
Comment (paulirish) [reply proto/lighthouse-result.proto]: > does any of this matter to consumers of the proto?

Not really. A LR-like client might be confused that when constructing a request and specifying something like `FormFactor: lhpb.ConfigSettings.EmulatedFormFactor.mobile`.

But that's not really an issue.

> ... except `none` was deprecated as an option, and that's the part that's _not_ explained by the comment :P

I added a small bit on that.

> Maybe (some version of) this comment makes more sense down on `emulated_form_factor` or the new `form_factor`?

Instead i just moved these 3 things to live closer. like a family. <3

**Review by exterkamp — APPROVED** [review]
LGTM mod that `emulated_form_factor` will still be used and populated in the PSI API in v7+, but it will have a new friend `form_factor` that will also be populated 😄

**Pull request merged**